spring-data-neo4j-rx/src/main/java/org/neo4j/springframework/data/core/cypher/DefaultStatementBuilder.java

```diff
@@ -218,6 +218,7 @@ public OngoingMatchAndUpdate remove(Node named, String... label) {
 	@Override
 	public OngoingReadingWithWhere where(Condition newCondition) {
 
+		Assert.notNull(newCondition, "The new condition must not be null.");
```

Comment:
Would it make more sense to move this assertion and the one below to the `ConditionBuilder`?